pkg/insight/utils.go

```diff
@@ -14,25 +14,41 @@
 
 package insight
 
-// import (
-// 	"time"
-
-// 	"github.com/pipe-cd/pipecd/pkg/model"
-// )
-
-// func NormalizeTime(from time.Time, step model.InsightStep) time.Time {
-// 	var formattedTime time.Time
-// 	switch step {
-// 	case model.InsightStep_DAILY:
-// 		formattedTime = time.Date(from.Year(), from.Month(), from.Day(), 0, 0, 0, 0, time.UTC)
-// 	case model.InsightStep_WEEKLY:
-// 		// Sunday in the week of rangeFrom
-// 		sunday := from.AddDate(0, 0, -int(from.Weekday()))
-// 		formattedTime = time.Date(sunday.Year(), sunday.Month(), sunday.Day(), 0, 0, 0, 0, time.UTC)
-// 	case model.InsightStep_MONTHLY:
-// 		formattedTime = time.Date(from.Year(), from.Month(), 1, 0, 0, 0, 0, time.UTC)
-// 	case model.InsightStep_YEARLY:
-// 		formattedTime = time.Date(from.Year(), 1, 1, 0, 0, 0, 0, time.UTC)
-// 	}
-// 	return formattedTime
-// }
+import (
+	"sort"
+	"time"
+
+	"github.com/pipe-cd/pipecd/pkg/model"
+)
+
+// NormalizeUnixTime ignores hour, minute, second and nanosecond
+func NormalizeUnixTime(t int64) int64 {
+	tt := time.Unix(t, 0)
+	return time.Date(tt.Year(), tt.Month(), tt.Day(), 0, 0, 0, 0, time.UTC).Unix()
+}
+
+func GroupDeploymentsByDaily(deployments []*model.InsightDeployment) [][]*model.InsightDeployment {
+	dailyDeployments := make(map[int64][]*model.InsightDeployment)
+
+	for _, d := range deployments {
+		t := NormalizeUnixTime(d.StartedAt)
```

Comment:
Btw, why do we need to normalize the time?

Reply:
I feel the same too 🤔

Reply:
This is because I want to group deployments by daily. `t` is only used by map key